Kernel_23/doc/Kernel_23/CGAL/Weighted_point_2.h

```diff
@@ -3,63 +3,206 @@ namespace CGAL {
 /*!
 \ingroup kernel_classes2
 
-An object of the class `Weighted_point_2` is a tuple of a 2D point and a scalar weight. 
+An object of the class `Weighted_point_2` is a tuple of a two-dimensional point and a scalar weight.
 
-
-\cgalHeading{Operators}
-
-The following operations can be applied on points: 
+Remember that `Kernel::RT` and `Kernel::FT` denote a
+`RingNumberType` and a `FieldNumberType`, respectively. For the kernel
+model `Cartesian<T>`, the two types are the same. For the
+kernel model `Homogeneous<T>`, `Kernel::RT` is equal
+to `T`, and `Kernel::FT` is equal to
+`Quotient<T>`.
 
 \sa `Point_2<Kernel>`
-\sa `Kernel::WeightedPoint_2` 
+
+\cgalModels `Kernel::WeightedPoint_2`
 
 */
 template< typename Kernel >
-class Weighted_point_2 {
+class Weighted_point_2
+{
 public:
+  /// \name Types
+  /// @{
 
+  /*!
+  An iterator for enumerating the
+  %Cartesian coordinates of a point.
+  */
+  typedef unspecified_type                Cartesian_const_iterator;
 
-/// \name Creation 
-/// @{
-  
-/*!
-introduces a weighted point with %Cartesian coordinates `(0,0)` and  weight `0`. 
-*/ 
-Weighted_point_2(const Origin &ORIGIN); 
+  /// @}
 
-/*!
-introduces a weighted point from point `p` and  weight `0`. 
-*/ 
-  Weighted_point_2(const Point_2<Kernel>& p); 
+  /// \name Creation
+  ///  Constructors from coordinates are provided for convenience, except
+  ///  that they are only from Cartesian coordinates, and with no weight, so as
+  ///  to avoid any potential ambiguity between the homogeneous weight and the
+  ///  power weight (it should be easy enough to pass a Point_2 explicitly in those
+  ///  cases).
```

Comment:
There are a lot of modifications in the documentation. They are probably good, but the previous version was reviewed, in the small feature. The new documentation should be peer-reviewed the same way.